src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1472,7 +1472,7 @@ public static IEnumerable<int> Range(int start, int count)
 
         private static IEnumerable<int> RangeIterator(int start, int count)
         {
-            for (int i = 0; i < count; i++) yield return start + i;
+            for (int end = start + count; start != end; start++) yield return start;
```

Comment:
> Is it ever possible for count to come in here as negative?

No, it's checked a few lines above:
https://github.com/dotnet/corefx/pull/2854/files#diff-95a14d3b4ae14339b623dd95c7537fd0R1469